### PR TITLE
Full-text Search

### LINKED ISSUE #13: Add free text search
**Is your feature request related to a problem? Please describe.**

There's a lot of information in this reference tool. It would be nice if we had the ability to search for terms that weren't only in titles.

**Describe the solution you'd like**

Perhaps something like this:

```
$ httpref --search auth

Authorisation
    This is a header that...

402
    This is returned when the request is unauthorised...
```

The search would be case insensitive (option?) and search though both headers and bodies.

Do we want to allow searches restricted to headers or status codes?

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by jesselang**
# Description

Added full-text search option for header, method, and status references.

Fixes #13

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

A single primitive unit test of the search function. If you have some specific cases you'd like covered, let me know.

- [x] Running existing tests
- [x] Created new tests

# Checklist:

- [x] My code has been linted (`make lint`)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased my branch to include the latest changes from `master`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +62,21 @@ The successful result of a PUT or a DELETE is often not a 200 OK but a 204 No Co
...
+## Full-text Search
+
+The `--search` option accepts a term to full-text search. This option is available for header, method, and status references.
+
+```
+$ httpref --search clear
+Clear-Site-Data
+  Clears browsing data (e.g. cookies, storage, cache) associated with the requesting
+   website.
+205
```
Comment (dnnrly) [inline README.md:74]: Should the `205` and `431` results contain `clear`?

**Comment by dnnrly** [comment]
Can you add a new test case to `acceptance.bats`? This will allow us to prove the plumbing still works in future tests.

Otherwise, very good contribution!

File: README.md (hunk 1 above)
Comment (jesselang) [reply README.md:74]: They do in their `Description`, but not in their `Name` or `Summary`.

**Comment by jesselang** [comment]
I've added some cases to `basic.bats`. Let me know how that strikes you. :smile:

File: README.md (hunk 1 above)
Comment (dnnrly) [reply README.md:74]: 🤦 

Of course it is. 😅

**Comment by dnnrly** [comment]
Yup, this is all good!

**Comment by jesselang** [comment]
Hey @dnnrly, would you be willing to label this PR as `hacktoberfest-accepted`? Many thanks!

**Pull request merged**